# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/webscheduling/WebTaskScheduler.cpp
# Commit: f441cb5e7e6e
# Full Hash: f441cb5e7e6e30d4a60b08ca99e56a29d3dc3a8b
# Author: Sean Feng <sefeng@mozilla.com>
# Date: 2025-04-01 09:40:15
# Regressor Bug: 1920115
# File Overlap Count: 2
# Description:
#   Bug 1920115 - Add the setup for WebTaskSchedulingState r=jjaschke,dom-core,arai
#   
#   Spec: https://wicg.github.io/scheduling-apis/#scheduling-state
#   
#   This patch add the setup for WebTaskSchedulingState so that it
# ==============================================================================

diff -r cfae5efaae37 -r f441cb5e7e6e dom/webscheduling/WebTaskScheduler.cpp
--- a/dom/webscheduling/WebTaskScheduler.cpp	Mon Mar 31 19:49:02 2025 +0000
+++ b/dom/webscheduling/WebTaskScheduler.cpp	Mon Mar 31 19:49:02 2025 +0000
@@ -22,6 +22,15 @@
   ImplCycleCollectionTraverse(aCallback, aQueue.Tasks(), aName, aFlags);
 }
 
+NS_IMPL_CYCLE_COLLECTION_CLASS(WebTaskSchedulingState)
+
+NS_IMPL_CYCLE_COLLECTION_TRAVERSE_BEGIN(WebTaskSchedulingState)
+  NS_IMPL_CYCLE_COLLECTION_TRAVERSE(mAbortSource, mPrioritySource);
+NS_IMPL_CYCLE_COLLECTION_TRAVERSE_END
+
+NS_IMPL_CYCLE_COLLECTION_UNLINK_BEGIN(WebTaskSchedulingState)
+  NS_IMPL_CYCLE_COLLECTION_UNLINK(mAbortSource, mPrioritySource);
+NS_IMPL_CYCLE_COLLECTION_UNLINK_END
 NS_IMPL_CYCLE_COLLECTION_CLASS(WebTask)
 
 NS_IMPL_CYCLE_COLLECTION_TRAVERSE_BEGIN(WebTask)